From 223b2291718495aa888df393a0f8756ae1d0e74a Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Thu, 24 Sep 2015 18:59:14 +0800
Subject: [PATCH 0517/5242] MLK-11623 ASoC: imx-cs42888: add 32k and 64k
 sample rate support

commit  98495d6ec8ec0572edd392a04b391820a141180f from
https://source.codeaurora.org/external/imx/linux-imx.git

When codec sysclk is 24576000, the sample rate ratio can be 128, 192,
256, 384, 512, 768, 1024. So 32k, 48k, 64k, 96k, 192k can be support.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
(cherry picked from commit b436254cd55dc0d2ebb6d2ca7ce5f58520ea5dbc)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-cs42888.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/sound/soc/fsl/imx-cs42888.c b/sound/soc/fsl/imx-cs42888.c
index 4153d39..1533805 100644
--- a/sound/soc/fsl/imx-cs42888.c
+++ b/sound/soc/fsl/imx-cs42888.c
@@ -80,11 +80,13 @@ static int imx_cs42888_surround_startup(struct snd_pcm_substream *substream)
 	int ret;
 
 	if (priv->mclk_freq == 24576000) {
-		support_rates[0] = 48000;
-		support_rates[1] = 96000;
-		support_rates[2] = 192000;
+		support_rates[0] = 32000;
+		support_rates[1] = 48000;
+		support_rates[2] = 64000;
+		support_rates[3] = 96000;
+		support_rates[4] = 192000;
 		constraint_rates.list = support_rates;
-		constraint_rates.count = 3;
+		constraint_rates.count = 5;
 
 		ret = snd_pcm_hw_constraint_list(runtime, 0, SNDRV_PCM_HW_PARAM_RATE,
 							&constraint_rates);
-- 
1.7.9.5

